src/merkle/merkle_tree.rs

```diff
@@ -115,11 +112,12 @@ impl MerkleTree {
     /// Returns an error if the specified index is not a valid leaf index for this tree.
     pub fn update_leaf(&mut self, index: u64, value: Word) -> Result<(), MerkleError> {
         let depth = self.depth();
-        if index >= 2u64.pow(depth) {
-            return Err(MerkleError::InvalidIndex(depth, index));
+        let node = NodeIndex::new(depth, index);
```

Comment:
nit: should we change the variable name here from `node` to `index`?

Reply:
Done!